docs/source/ttnn/ttnn/adding_new_ttnn_operation.rst

```diff
@@ -7,174 +7,98 @@ Adding New ttnn Operation
+A concrete example of a device operation can be found in `ttnn/cpp/ttnn/operations/examples/example/device`
 
-.. code-block:: cpp
+`ttnn/cpp/ttnn/operations/examples/example/device/example_device_operation.hpp`:
 
-    m_tensor.def("<new_operation>", &<new_operation>, py::arg("input_tensor").noconvert(), py::arg("some_arg").noconvert(), R"doc(
-        <NewOperation> runs new operation on input tensor.
+.. literalinclude::  examples/example/device/example_device_operation.hpp
 
-        .. csv-table::
-            :header: "Argument", "Description", "Data type", "Valid range", "Required"
+`ttnn/cpp/ttnn/operations/examples/example/device/example_device_operation.cpp`:
 
-            "input_tensor", "Input tensor", "Tensor", "Tensor of shape [W0, Z0, Y0, X0]", "Yes"
-            "some_arg", "Some arg", "bool", "Some arg to do some stuff in new operation", "Yes"
-    )doc");
+.. literalinclude::  examples/example/device/example_device_operation.cpp
 
+`ttnn/cpp/ttnn/operations/examples/example/device/single_core_program_factory.cpp`:
 
+.. literalinclude::  examples/example/device/single_core_program_factory.cpp
 
-Adding a unit test
-------------------
+`ttnn/cpp/ttnn/operations/examples/example/device/multi_core_program_factory.cpp`:
 
-Add `tests/ttnn/unit_tests/ttl/test_<new_operation>.py`:
+.. literalinclude::  examples/example/device/multi_core_program_factory.cpp
 
-.. code-block:: python
 
-    import pytest
-    import torch
-    import ttnn
+Step 2: Implement the operation in C++
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    from tests.ttnn.utils_for_testing import assert_with_pcc
+In order to add a new operation, add the following file:
 
-    @pytest.mark.parametrize("height", [32])
-    @pytest.mark.parametrize("width", [32])
-    def test_<new_operation>(device, height, width):
-        torch.manual_seed(0)
+`ttnn/cpp/ttnn/operations/<category>/<operation_name>/<operation_name>.hpp`
 
-        torch_input_tensor = torch.rand(1, 1, height, width)
-        torch_output_tensor = torch.exp(torch_input_tensor)
+A concrete example:
 
-        input_tensor = ttnn.from_torch(torch_input_tensor, device=device)
-        output_tensor = ttnn.experimental.tensor.<new_operation>(input_tensor)
+`ttnn/cpp/ttnn/operations/examples/example/example.hpp`:
 
-        output_tensor = ttnn.to_torch(output_tensor)
+.. literalinclude::  examples/example/example.hpp
 
-        assert_with_pcc(torch_output_tensor, output_tensor)
 
+Python Implementation
+---------------------
 
+Step 1: Add Python binding
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Adding a sweep test
--------------------
+In order to add a python binding for the operation, follow the directory structure shown below:
 
-Add `tests/ttnn/sweep_tests/sweeps/ttl_<new_operation>.py`:
+`ttnn/python/ttnn/operations/<category>/<operation_name>/<operation_name>_pybind.hpp`
+`ttnn/python/ttnn/operations/<category>/<category>_pybind.hpp`
 
-.. code-block:: python
+A concrete example:
```

Comment:
maybe lets add some notes about consistency? not sure if here or in general.
we want ops to be consistent.
e.g. 
kw_only memory_config = None
kw_only queue_id in python and overload in c++
kw_only optional output tensor/s
kw_only dtype 
Specific names, specific defaults EXCEPT when not possible and something else is needed.

Reply:
will do in the next PR